sklearn/preprocessing/tests/test_data.py

```diff
@@ -1201,53 +1210,59 @@ def test_quantile_transform_check_error():
                           [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]])
     X_neg = sparse.csc_matrix(X_neg)
 
-    assert_raises_regex(ValueError, "Invalid value for 'n_quantiles': 0.",
-                        QuantileTransformer(n_quantiles=0).fit, X)
-    assert_raises_regex(ValueError, "Invalid value for 'subsample': 0.",
-                        QuantileTransformer(subsample=0).fit, X)
-    assert_raises_regex(ValueError, "The number of quantiles cannot be"
-                        " greater than the number of samples used. Got"
-                        " 1000 quantiles and 10 samples.",
-                        QuantileTransformer(subsample=10).fit, X)
+    with pytest.raises(ValueError, match="Invalid value for "
+                                         "'n_quantiles': 0."):
+        QuantileTransformer(n_quantiles=0).fit(X)
+    with pytest.raises(ValueError, match="Invalid value for "
+                                         "'subsample': 0."):
+        QuantileTransformer(subsample=0).fit(X)
+    with pytest.raises(ValueError, match="The number of quantiles "
+                                         "cannot be greater than "
+                                         "the number of samples "
+                                         "used. Got 1000 quantiles "
+                                         "and 10 samples."):
+        QuantileTransformer(subsample=10).fit(X)
 
     transformer = QuantileTransformer(n_quantiles=10)
-    assert_raises_regex(ValueError, "QuantileTransformer only accepts "
-                        "non-negative sparse matrices.",
-                        transformer.fit, X_neg)
+    with pytest.raises(ValueError, match="QuantileTransformer only accepts "
```

Comment:
Same comments regarding the error message below. Try to move them out of `raises` and when possible you can loop.